### PR TITLE
[workloadmeta] Remove dedicated namespace collection config

### PR DESCRIPTION
**Pull request by jennchenn**
### What does this PR do?

Remove the config to enable collection of namespace information in workloadmeta in favor of the generic config introduced in https://github.com/DataDog/datadog-agent/pull/25901.

### Motivation

We will be migrating references to the namespace-specific store to use the generic kubernetes metadata collection types. In the meantime, to avoid exposing temporary configurations, we want to configure namespace collection using the generic kubernetes metadata collection config.  

### Describe how to test/QA your changes

1. Deploy the agent and cluster agent with namespace resource collection enabled
```
...
clusterAgent:
  enabled: true
  env:
    - name: DD_CLUSTER_AGENT_KUBE_METADATA_COLLECTION_ENABLED
      value: "true"
    - name: DD_CLUSTER_AGENT_KUBE_METADATA_COLLECTION_RESOURCES
      value: "namespaces daemonsets"
```

2. Check the content of workloadmeta of the DCA using `agent workload-list -v`; daemonsets metadata are collected as `kubernetes_metadata` entities (all `kubernetes_metadata` entities should only be daemonset metadata)
```
=== Entity kubernetes_metadata sources(merged):[kubeapiserver] id: daemonsets/default/datadog-agent ===
----------- Entity ID -----------
Kind: kubernetes_metadata ID: daemonsets/default/datadog-agent

----------- Entity Meta -----------
Name: datadog-agent
Namespace: default
Annotations: meta.helm.sh/release-name:datadog-agent meta.helm.sh/release-namespace:default 
Labels: app.kubernetes.io/component:agent app.kubernetes.io/instance:datadog-agent app.kubernetes.io/managed-by:Helm app.kubernetes.io/name:datadog-agent app.kubernetes.io/version:7 helm.sh/chart:datadog-3.59.5 
----------- Resource -----------
apps/v1, Resource=daemonsets===
```

Namespace metadata are collected as `kubernetes_namespace` entities only
```
=== Entity kubernetes_namespace sources(merged):[kubeapiserver] id: annotated-namespace ===
----------- Entity ID -----------
Kind: kubernetes_namespace ID: annotated-namespace

----------- Entity Meta -----------
Name: annotated-namespace
Namespace: 
Annotations: kubectl.kubernetes.io/last-applied-configuration:{"apiVersion":"v1","kind":"Namespace","metadata":{"annotations":{"related_email":"express@sunday"},"labels":{"injectSidecarNs":"true","related_team":"contp"},"name":"annotated-namespace"}} related_email:express@sunday 
Labels: injectSidecarNs:true kubernetes.io/metadata.name:annotated-namespace related_team:contp 
===
```

### CODE REVIEW

File: comp/core/workloadmeta/collectors/internal/kubeapiserver/kubeapiserver.go [hunk 1]
```diff
@@ -46,8 +46,15 @@ func storeGenerators(cfg model.Reader) []storeGenerator {
 		generators = append(generators, newDeploymentStore)
 	}
 
-	if cfg.GetBool("kubernetes_namespace_collection_enabled") {
-		generators = append(generators, newNamespaceStore)
+	// TODO: Remove this once we migrate references to the namespace store to use generic collection
+	if cfg.GetBool("cluster_agent.kube_metadata_collection.enabled") {
+		resources := cfg.GetStringSlice("cluster_agent.kube_metadata_collection.resources")
+		for _, resource := range resources {
+			if resource == "namespaces" {
```
Comment (davidor) [inline comp/core/workloadmeta/collectors/internal/kubeapiserver/kubeapiserver.go:53]: nit: you can use `slices.Contains` to simplify some code.

**Review by davidor — APPROVED** [review]
We'll also need to disable namespace collection in the generic collector when `cluster_agent.kube_metadata_collection.enabled` is true and configured with the `namespaces` resource. Otherwise, we'll have 2 collectors doing the same.
But I think we can do that in a separate PR.

**Comment by adel121** [comment]
I opened [this](https://github.com/DataDog/datadog-agent/pull/26170) PR on top of the current branch, and I merged it.

This will help avoiding registering both collectors for namespace collection at the same time.

**Review by adel121 — APPROVED** [review]

**Review by dustmop — APPROVED** [review]
LGTM

**Pull request merged**